src/VecSim/algorithms/brute_force/brute_force_multi.h

```diff
@@ -44,14 +45,16 @@ class BruteForceIndex_Multi : public BruteForceIndex {
         assert(!"should have found the old id");
     }
 
-    inline vecsim_stl::abstract_priority_queue<float, labelType> *getNewPriorityQueue() override {
+    inline vecsim_stl::abstract_priority_queue<DistType, labelType> *
+    getNewPriorityQueue() override {
```

Comment:
Are you sure? I found [this](https://stackoverflow.com/questions/17667098/inline-template-function) question that says otherwise.

Reply:
They actually say the same thing
Inline **cant** be omitted in case this is **template specialization** because this is the same as declaring a regular function - 
its symbol will be placed in the text section
Templates, with no specialization, are generated as weak symbols (using optimization the compiler is allowed to decide to inline them, meaning that they will not appear in the symbol table at all), which is basically what inline does...
both for templates and for inline functions
1. The compiler must know the definition in compile time - this is why we must put templates in the header
2. Using appropriate optimization the compiler might choose to inline these function